cmds/core/chmod/chmod.go

```diff
@@ -135,54 +148,61 @@ func calculateMode(modeString string) (mode os.FileMode, octval uint64, mask uin
 	return mode, octval, mask, nil
 }
 
-func chmod(recursive bool, reference string, args ...string) (fs.FileMode, error) {
+func (c *cmd) run(args ...string) error {
 	var mode os.FileMode
 	if len(args) < 1 {
-		return mode, errBadUsage
+		return errBadUsage
 	}
 
-	if len(args) < 2 && reference == "" {
-		return mode, errBadUsage
+	if len(args) < 2 && c.reference == "" {
+		return errBadUsage
 	}
 
 	var (
-		err          error
 		octval, mask uint64
 		fileList     []string
 	)
 
-	if reference != "" {
-		fi, err := os.Stat(reference)
+	if c.reference != "" {
+		fi, err := os.Stat(c.reference)
 		if err != nil {
-			return 0, fmt.Errorf("bad reference file: %w", err)
+			return fmt.Errorf("bad reference file: %w", err)
 		}
 		mask = special
 		mode = fi.Mode()
 		fileList = args
 	} else {
 		var err error
 		if mode, octval, mask, err = calculateMode(args[0]); err != nil {
-			return mode, err
+			return err
 		}
 		fileList = args[1:]
 	}
 
+	var finalErr error
+
 	for _, name := range fileList {
-		if recursive {
-			err := filepath.Walk(name, func(path string, info os.FileInfo, err error) error {
-				mode, err = changeMode(path, mode, octval, mask)
+		if c.recursive {
+			err := filepath.Walk(name, func(path string, _ os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+				err = changeMode(path, mode, octval, mask)
 				return err
 			})
 			if err != nil {
-				return mode, err
+				finalErr = err
+				fmt.Fprintln(c.stderr, err)
 			}
 		} else {
-			if mode, err = changeMode(name, mode, octval, mask); err != nil {
-				return mode, err
+			err := changeMode(name, mode, octval, mask)
```

Comment:
would it be worth using errors.Join here, so we get them all? That might require moving Go forward but ... ?